src/main/webapp/resources/js/viewer/controller.js

```diff
@@ -688,6 +719,15 @@ function Controller(bookID, accessible_modes, canvasID, regionColors, colors, gl
 	}
 
 	this.exportPageXML = function (_page = _currentPage) {
+		if(_segmentation[_currentPage] != null) {
```

Comment:
Looks to be the same code as in lines 493-502. Should be encapsulated inside function.